### PR TITLE
interfaces: we actually expect snaps to auto-connect to desktop

### PR DESCRIPTION
**Pull request by pedronis**
originally desktop was simply allow auto-connect, making the change to not auto-connect on UC is not that relevant because ultimately there will need to be a snap providing it and that snap will set policy if allowed from its snap-declaration

instead let auto-connection only happen with an implicit slot

### CODE REVIEW

**Review by alexmurray — APPROVED** [review]
LGTM (once this is merged we can then update review-tools accordingly)

**Review by ZeyadYasser — APPROVED** [review]
LGTM, Thank you!

**Review by jhenstridge — APPROVED** [review]
This seems reasonable, although I'm not sure what effect it would have in practice.

We did use `on-classic` in the snap declaration for the `ubuntu-desktop-session:desktop` slot, but it wasn't under the belief that it was needed to override the base declaration.

Rather it was to avoid support problems if people happened to install the snap on a classic system, where having multiple connection candidates for the interface would prevent any auto-connection.

**Review by andrewphelpsj — APPROVED** [review]
Looks good to me, thanks!

**Comment by pedronis** [comment]
@jhenstridge as discussed elsewhere the material effect of this mostly related to the store permissions' UI

**Pull request merged**
